src/app/TraitReflections.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Backpack\PageManager\app;
+
+trait TraitReflections
+{
+    public function checkForTemplatesAndUniquePagesNotDistinct()
+    {
+        if (config('backpack.pagemanager.page_model_class') != config('backpack.pagemanager.unique_page_model_class')) {
+            return;
+        }
+
+        $uniquePages = collect($this->getUniquePages())->pluck('name');
+        $templates = collect($this->getTemplates())->pluck('name');
+
+        if ($uniquePages->intersect($templates)->isNotEmpty()) {
+            throw new \Exception('Templates and unique pages should not have the same function names when same model class is used.');
+        }
+    }
+
+    /**
+     * Get all defined unique pages.
+     */
+    public function getUniquePages()
+    {
+        $pages_trait = new \ReflectionClass('App\UniquePages');
+        $pages = $pages_trait->getMethods(\ReflectionMethod::IS_PRIVATE);
```

Comment:
All the private methods? That's odd.

Reply:
Well that's exactly how the current page templates work at the moment...

All private methods in the Trait are defining one single template (or here on unique page)